Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -428,110 +599,219 @@ void DiscusMultipleScatteringCorrection::exec() {
   // set the output property
   setProperty("OutputWorkspace", wsgroup);
 
-  if (g_log.is(Kernel::Logger::Priority::PRIO_WARNING)) {
+  if (g_log.is(Kernel::Logger::Priority::PRIO_INFORMATION)) {
     for (auto &kv : m_attemptsToGenerateInitialTrack)
-      g_log.warning() << "Generating initial track required " + std::to_string(kv.first) + " attempts on " +
-                             std::to_string(kv.second) + " occasions.\n";
-    g_log.warning() << "Calls to interceptSurface= " + std::to_string(m_callsToInterceptSurface) + "\n";
+      g_log.information() << "Generating initial track required " + std::to_string(kv.first) + " attempts on " +
+                                 std::to_string(kv.second) + " occasions.\n";
+    g_log.information() << "Calls to interceptSurface= " + std::to_string(m_callsToInterceptSurface) + "\n";
   }
 }
 
+/**
+ * Generate a list of the k and w points where calculation results are required. The w points are expressed
+ * as bin indices and values.
+ * The special bin index value -1 means calculate results for all w bins in the innermost calculation loop using a
+ * single set of simulated tracks
+ * @param efixed The fixed energy (or zero if an elastic calculation)
+ * @param xPoints The x points either in momentum (elastic) or energy transfer (inelastic)
+ */
+std::vector<std::tuple<double, int, double>>
+DiscusMultipleScatteringCorrection::generateInputKOutputWList(const double efixed, const std::vector<double> &xPoints) {
+  std::vector<std::tuple<double, int, double>> kInW;
+  const double kFixed = toWaveVector(efixed);
+  if (m_EMode == DeltaEMode::Elastic) {
+    int index = 0;
+    std::transform(xPoints.begin(), xPoints.end(), std::back_inserter(kInW), [&index](double d) {
+      auto t = std::make_tuple(d, index, 0.);
+      index++;
+      return t;
+    });
+  } else {
```

Comment:
Return kInW from this if branch and remove the else to remove a layer of nesting

Reply:
As a point of style I prefer having the single point of return in this case because the different branches are genuine alternatives ie the first "if branch" isn't some kind of pre-validation before proceeding to the real work in the later branches. If I chose in future to post-process the result (eg filter it or sort it) I can also do it once at the moment.
If there's some kind of general rule either in Mantid or the wider C++ world that I'm going against though I'm happy to be told otherwise

Reply:
I mean it's more like emacs vs vim in the C++ world so I'm not going to get too involved on this one haha.

My concern is the nested logic in the else branch. It feels like there are 2-3 private methods rolled into a single method for INDIRECT / DIRECT / ELASTIC that could be unrolled out (the one liner for DIRECT should stay at the moment)

Reply:
It feels like those lines belong together. The Direct case is actually split into two different approaches depending on the `SimulateEnergiesIndependently` flag (one in the for loop on L628 and one outside on L624). I think seeing all the different tuples that get put into the kInW vector in one function is also useful for seeing what this vector means\does.